### PR TITLE
Describe the bytesize cost of hints

### LINKED ISSUE #962: Hints aren't free
Enabling a client hint adds a number of bytes to every request.  With HPACK, that might eventually be as little as 1 byte, but this isn't free.  The draft should recognize this cost and caution against turning everything into a hint.

### PR DESCRIPTION
**Pull request by yoavweiss**
Closes #962 

@martinthomson - Took a first stab at describing the cost of hints and the fact that servers should not overuse them. Let me know what you think.

/cc @igrigorik

### CODE REVIEW

**Comment by yoavweiss** [comment]
Added that features MAY restrict hints to certain destinations to close #963

File: draft-ietf-httpbis-client-hints.md [hunk 1]
```diff
@@ -168,6 +168,12 @@ Implementers ought to consider both user and server controlled mechanisms and po
 
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
+# Other Considerations
+
+While HTTP header compression schemes reduce the cost of adding HTTP header fields, sending Client Hints to the server incurs an increase in request byte size.
+Servers SHOULD take that into account when opting in to receive Client Hints, and SHOULD NOT opt-in to receive hints unless they are to be used for content adaptation purposes.
+
+Due to request byte size increase, features relying on this document to define Client Hints MAY consider restricting sending those hints to certain request destinations, where they are more likely to be useful.
```
Comment (igrigorik) [inline draft-ietf-httpbis-client-hints.md]: Worth linking to Fetch dfn of `destination`? Figure most folks won't make the connection.

Comment (yoavweiss) [reply draft-ietf-httpbis-client-hints.md]: Yup!

File: draft-ietf-httpbis-client-hints.md [hunk 2]
```diff
@@ -168,6 +176,12 @@ Implementers ought to consider both user and server controlled mechanisms and po
 
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
+# Other Considerations
```
Comment (martinthomson) [inline draft-ietf-httpbis-client-hints.md]: Maybe

```suggestion
# Cost of Sending Hints
```

**Comment by yoavweiss** [comment]
@martinthomson Thanks for reviewing! :)

Can you commit this? (I don't have permissions)

**Pull request merged**